cmd/controller-gen/main.go

```diff
@@ -17,197 +17,321 @@ package main
+	if name != "" {
+		out.WriteString(name)
+	}
+	out.WriteRune('=')
+
+	out.WriteRune('<')
+	out.WriteString(arg.TypeString())
+	out.WriteRune('>')
 
-	return cmd
+	if arg.Optional {
+		out.WriteString("]")
+	}
 }
 
-func newCRDCmd() *cobra.Command {
-	g := &crdgenerator.Generator{}
+// optionHelp assembles help for a given named option (specifed via marker).
+func optionHelp(def *markers.Definition, out *strings.Builder) {
+	out.WriteString(def.Name)
+	if def.Empty() {
+		return
+	}
+
+	first := true
+	for argName, arg := range def.Fields {
+		fieldHelp(argName, arg, first, out)
+		first = false
+	}
+}
+
+// allOptionsHelp documents all accepted options
+func allOptionsHelp(reg *markers.Registry, descTarget bool, sortFunc func(string, string) bool) string {
+	out := &strings.Builder{}
+	allMarkers := reg.AllDefinitions()
+	sort.Slice(allMarkers, func(i, j int) bool {
+		// sort grouping marker options together
+		iName, jName := allMarkers[i].Name, allMarkers[j].Name
+		return sortFunc(iName, jName)
+	})
+	for _, def := range allMarkers {
+		out.WriteString("  ")
+		optionHelp(def, out)
+		if descTarget {
+			switch def.Target {
+			case markers.DescribesPackage:
+				out.WriteString(" (package)")
+			case markers.DescribesType:
+				out.WriteString(" (type)")
+			case markers.DescribesField:
+				out.WriteString(" (struct field)")
+			}
+		}
+		out.WriteRune('\n')
+	}
+	return out.String()
+}
 
+func compOptionsForSort(iName, jName string) bool {
+	iParts := strings.Split(iName, ":")
+	jParts := strings.Split(jName, ":")
+
+	iGen := ""
+	iRule := ""
+	jGen := ""
+	jRule := ""
+	
+	switch len(iParts) {
+	case 1:
+		iGen = iParts[0]
+	// two means a default output rule, so ignore
+	case 2:
+		iRule = iParts[1]
+	case 3:
+		iGen = iParts[1]
+		iRule = iParts[2]
+	}
+	switch len(jParts) {
+	case 1:
+		jGen = jParts[0]
+	// two means a default output rule, so ignore
+	case 2:
+		jRule = jParts[1]
+	case 3:
+		jGen = jParts[1]
+		jRule = jParts[2]
+	}
+
+	if iGen != jGen {
+		return iGen > jGen
+	}
+
+	return iRule < jRule
+}
+
+func main() {
+	printMarkersOnly := false
 	cmd := &cobra.Command{
-		Use:   "crd",
-		Short: "Generates CRD manifests",
-		Long: `Generate CRD manifests from the Type definitions in Go source files.
-Usage:
-# controller-gen crd [--domain k8s.io] [--root-path input_dir] [--output-dir output_dir]
+		Use:   "controller-gen",
+		Short: "Generate Kubernetes API extension resources and code.",
+		Long: "Generate Kubernetes API extension resources and code.",
+		Example: `	# Generate RBAC manifests and crds for all types under apis/,
+	# outputting crds to /tmp/crds and everything else to stdout
+	controller-gen rbac crd paths=./apis/... output:crd:dir=/tmp/crds output:stdout
+
+	# Generate deepcopy implementations for a particular file
+	controller-gen deepcopy paths=./apis/v1beta1/some_types.go
+
+	# Run all the generators for a given project
+	controller-gen paths=./apis/...
```

Comment:
is `all` supported ? because that is the arg which is used by all the kubebuilder projects.

Reply:
not specifying any generators means "all"